remix/app/services/auth.server.ts

```diff
@@ -0,0 +1,44 @@
+import { Authenticator } from "remix-auth";
+import { sessionStorage } from "./session.server";
+import { GoogleStrategy, SocialsProvider } from "remix-auth-socials";
+import type { UserProfile } from "~/types";
+import { db } from "~/utils/db.server";
+
+export const authenticator = new Authenticator(sessionStorage);
+
+async function handleSocialAuthCallBack({ profile }: any) {
+  const userProfile: UserProfile = profile?._json;
+  if (userProfile) {
+    try {
+      await db.user.upsert({
+        where: {
+          email: userProfile.email,
+        },
+        update: {
+          name: userProfile.name,
+          avatarUrl: userProfile.picture,
+        },
+        create: {
+          email: userProfile.email,
+          name: userProfile.name,
+          avatarUrl: userProfile.picture,
+        },
+      });
+    } catch (err) {
+      console.log(err);
+    }
+  }
+  return profile;
+}
+
+authenticator.use(
```

Comment:
Shouldn't this be initialized in a config and upon booting the application? If no credentials are found, the app should not start.

Reply:
`authenticator.use(` yes, that was my mistake it should initialized in the config, at the time when I initialized it I was focus on how to make work. so now I will move it to config.